amp/WEB-INF/src/org/digijava/module/aim/util/SectorUtil.java

```diff
@@ -1214,12 +1238,109 @@ public static List<AmpActivityVersion> getActivitiesForSector(Long id) {
         }
         return activities;
     }
-    
+
     public static List<String> getAllSectorColumnNames() {
         // not much benefit from generating names
         return Arrays.asList(
                 ColumnConstants.PRIMARY_SECTOR, ColumnConstants.PRIMARY_SECTOR_SUB_SECTOR, ColumnConstants.PRIMARY_SECTOR_SUB_SUB_SECTOR,
                 ColumnConstants.SECONDARY_SECTOR, ColumnConstants.SECONDARY_SECTOR_SUB_SECTOR, ColumnConstants.SECONDARY_SECTOR_SUB_SUB_SECTOR,
                 ColumnConstants.TERTIARY_SECTOR, ColumnConstants.TERTIARY_SECTOR_SUB_SECTOR, ColumnConstants.TERTIARY_SECTOR_SUB_SUB_SECTOR);
     }
+
+    //region Sector Mapping
+    /**
+     * Returns all sector mappings
+     */
+    public static Collection getAllSectorMappings() {
+        String queryString = null;
+        Session session = null;
+        Collection col = null;
+        Query qry = null;
+
+        try {
+            session = PersistenceManager.getSession();
+            queryString = "select asm from " + AmpSectorMapping.class.getName() + " asm";
+            qry = session.createQuery(queryString);
+            col = qry.list();
+            session.flush();
+        } catch (Exception ex) {
+            logger.error("Unable to get sectors mappings from database " + ex.getMessage());
+            ex.printStackTrace(System.out);
+        }
+        return col;
+    }
+
+    /**
+     * adds an AmpSectorMapping
+     */
+    public static void createSectorMapping(AmpSectorMapping sectorMapping) throws DgException {
+        Session session = null;
+        if (searchSectorMapping(sectorMapping.getSrcSector().getAmpSectorId(), sectorMapping.getDstSector().getAmpSectorId()) == null) {
+            try {
+                session = PersistenceManager.getRequestDBSession();
+                session.saveOrUpdate(sectorMapping);
+                session.flush();
+            } catch (Exception ex) {
+                logger.error("Unable to save a sector mapping " + ex.getMessage());
+                throw new DgException(ex);
+            }
+        }
+    }
+
+    public static void createSectorMappings(List<AmpSectorMapping> sectorMappings) throws DgException {
+        Session session = null;
+        Collection<AmpSectorMapping> existingMappings = getAllSectorMappings();
+        try {
+            session = PersistenceManager.getRequestDBSession();
+
+            for (AmpSectorMapping em : existingMappings) {
+                session.delete(em);
+            }
+            session.flush();
+            for (AmpSectorMapping asm : sectorMappings) {
+                session.save(asm);
+            }
+            session.flush();
```

Comment:
We can combine the flush operations to one at the end

Reply:
I made it but it didn't work. When I used only one flush at the end, I receive SQL error by dyplicate rows, but with 2 flush the deletion was made succefully and the new mappings were saved ok

Reply:
Okay